diff -Naur gnome-session-42.0/gnome-session/gnome-session.in gnome-session-42.0-patched/gnome-session/gnome-session.in
--- gnome-session-42.0/gnome-session/gnome-session.in	2022-03-21 15:13:31.651419000 -0500
+++ gnome-session-42.0-patched/gnome-session/gnome-session.in	2022-08-07 15:29:48.747954696 -0500
@@ -13,6 +13,69 @@
   fi
 fi
 
+# Get blocaled's value for LANG over the system dbus. This is the system wide setting in GNOME:
+gnome_lang_sys="$(busctl get-property org.freedesktop.locale1 /org/freedesktop/locale1 org.freedesktop.locale1 Locale | cut -d\" -f2 | cut -d= -f2)"
+
+# Get the AccountsService value for LANG over the system dbus. This is the user setting in GNOME:
+gnome_lang_usr="$(busctl get-property org.freedesktop.Accounts /org/freedesktop/Accounts/User${UID} org.freedesktop.Accounts.User Language | cut -d\" -f2)"
+
+# Source the standard profile scripts in a gnome-session.
+# This code was copied from KDE's "wayland-session" script.
+# Copyright (C) 2015-2016 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
+# This file is extracted from kde-workspace (kdm/kfrontend/genkdmconf.c)
+# Copyright (C) 2001-2005 Oswald Buddenhagen <ossi@kde.org>
+# Modified and added to gnome's "gnome-session" script by Bob Funk <bobfunk11@gmail.com>
+# Note that the respective logout scripts are not sourced.
+case $SHELL in
+  */bash)
+    [ -z "$BASH" ] && exec $SHELL $0 "$@"
+    set +o posix
+    [ -f /etc/profile ] && . /etc/profile
+    [ -n "$gnome_lang_sys" ] && export LANG=$gnome_lang_sys
+    if [ -f $HOME/.bash_profile ]; then
+      . $HOME/.bash_profile
+    elif [ -f $HOME/.bash_login ]; then
+      . $HOME/.bash_login
+    elif [ -f $HOME/.profile ]; then
+      . $HOME/.profile
+    fi
+    [ -n "$gnome_lang_usr" ] && export LANG=$gnome_lang_usr
+  ;;
+  */zsh)
+    [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
+    [ -d /etc/zsh ] && zdir=/etc/zsh || zdir=/etc
+    [ -n "$gnome_lang_sys" ] && export LANG=$gnome_lang_sys
+    zhome=${ZDOTDIR:-$HOME}
+    # zshenv is always sourced automatically.
+    [ -f $zdir/zprofile ] && . $zdir/zprofile
+    [ -f $zhome/.zprofile ] && . $zhome/.zprofile
+    [ -f $zdir/zlogin ] && . $zdir/zlogin
+    [ -f $zhome/.zlogin ] && . $zhome/.zlogin
+    [ -n "$gnome_lang_usr" ] && export LANG=$gnome_lang_usr
+    emulate -R sh
+  ;;
+  */csh|*/tcsh)
+    # [t]cshrc is always sourced automatically.
+    # Note that sourcing csh.login after .cshrc is non-standard.
+    wlsess_tmp=`mktemp /tmp/wlsess-env-XXXXXX`
+    $SHELL -c "if (-f /etc/csh.login) source /etc/csh.login; if (-f ~/.login) source ~/.login; /bin/sh -c 'export -p' >! $wlsess_tmp"
+    . $wlsess_tmp
+    rm -f $wlsess_tmp
+  ;;
+  */fish)
+    xsess_tmp=`mktemp /tmp/xsess-env-XXXXXX`
+    $SHELL --login -c "/bin/sh -c 'export -p' > $xsess_tmp"
+    . $xsess_tmp
+    rm -f $xsess_tmp
+  ;;
+  *) # Plain sh, ksh, and anything we do not know.
+    [ -f /etc/profile ] && . /etc/profile
+    [ -n "$gnome_lang_sys" ] && export LANG=$gnome_lang_sys
+    [ -f $HOME/.profile ] && . $HOME/.profile
+    [ -n "$gnome_lang_usr" ] && export LANG=$gnome_lang_usr
+  ;;
+esac
+
 SETTING=$(G_MESSAGES_DEBUG='' gsettings get org.gnome.system.locale region)
 REGION=${SETTING#\'}
 REGION=${REGION%\'}
